http/src/main/scala/org/http4s/blaze/http/http2/FlowStrategy.scala

```diff
@@ -20,16 +20,20 @@ trait FlowStrategy {
     *
     * @note This must not mutate the [[SessionFlowControl]] or the [[StreamFlowWindow]] in any way.
     *
-    * @param session the session [[SessionFlowControl]]
     * @param stream the stream [[StreamFlowWindow]]
     * @return the number of bytes to update the session and stream flow window with.
     */
-  def checkStream(session: SessionFlowControl, stream: StreamFlowWindow): Increment
+  def checkStream(stream: StreamFlowWindow): Increment
 }
 
 object FlowStrategy {
   final class Increment private[FlowStrategy](val session: Int, val stream: Int) {
-    override def toString: String = s"Increment($session, $stream)"
+    override def toString = s"Increment($session, $stream)"
+    override def hashCode(): Int = session + stream
```

Comment:
This seems like a weird hash code.  Is it because these values are fairly random anyway?

Any reason this isn't just a case class?

Reply:
It's only so I have something to go with the overridden equals. It's not a case class because I wanted to implement the `apply` method, or more specifically, I didn't want the synthesized `apply` so I could optimize the common case of `Empty`. Maybe that's not a great reason: I also dislike the noise.